wdl-doc/tests/doc.rs

```diff
@@ -62,31 +78,55 @@ async fn main() {
     // repopulating it with the generated docs (at `tests/codebase/docs/`).
     if env::var("BLESS").is_ok() {
         if docs_dir.exists() {
-            fs::remove_dir_all(docs_dir).unwrap();
+            fs::remove_dir_all(&docs_dir).unwrap();
         }
-        fs::create_dir_all(docs_dir).unwrap();
-        copy_dir_all(test_dir.join("docs"), docs_dir).unwrap();
+        fs::create_dir_all(&docs_dir).unwrap();
+        copy_dir_all(test_dir.join("docs"), &docs_dir).unwrap();
 
         println!("Blessed docs");
         exit(0);
     }
 
-    // Compare the generated docs with the expected output
-    // For now, check that the paths exist as expected.
-    // TODO: check HTML content.
+    // Compare the generated docs with the expected output.
+    // Recursively read the contents of the `tests/codebase/docs` directory
+    // and compare them with the contents of the `tests/output_docs` directory.
+    // If the contents are different, print the differences and exit with a
+    // non-zero exit code.
     let mut success = true;
-    for entry in fs::read_dir(docs_dir).unwrap() {
-        let entry = entry.unwrap();
-        let path = entry.path();
-        // Normalize the path to be relative to the `docs` directory
-        // regardless of OS path separator.
-        let expected_path = test_dir
-            .join("docs")
-            .join(path.strip_prefix(docs_dir).unwrap());
-        if !expected_path.exists() {
-            eprintln!("Expected path does not exist: {}", expected_path.display());
+    for file_name in read_dir_recursively(&test_dir.join("docs")).unwrap() {
+        let expected_file = docs_dir.join(file_name.strip_prefix(test_dir.join("docs")).unwrap());
+        if !expected_file.exists() {
+            println!("Missing file: {}", expected_file.display());
             success = false;
+            continue;
+        }
+
+        if expected_file.extension().and_then(|e| e.to_str()) == Some("svg") {
+            // Ignore image files
+            continue;
         }
+
+        // TODO: snapshotting the HTML/CSS files is not a good test,
```

Comment:
Does this mean we should remove the existing test?

Reply:
Also, this is being tracked in this issue https://github.com/stjude-rust-labs/wdl/issues/453

Reply:
I would say let's turn it on again for now. I think that's better than having something that _looks_ like snapshot testing but is, in fact, just testing that the files are present. If it becomes a larger issue, we will be incentivized to change it in the future.